docs/standard/basetypes/alternation.md

```diff
@@ -0,0 +1,328 @@
+
+```csharp
+using System;
+using System.Text.RegularExpressions;
+
+public class Example
+{
+   public static void Main()
+   {
+      // Regular expression using character class.
+      string pattern1 = @"\bgr[ae]y\b";
+      // Regular expression using either/or.
+      string pattern2 = @"\bgr(a|e)y\b";
+
+      string input = "The gray wolf blended in among the grey rocks.";
+      foreach (Match match in Regex.Matches(input, pattern1))
+         Console.WriteLine("'{0}' found at position {1}", 
+                           match.Value, match.Index);
+      Console.WriteLine();
+      foreach (Match match in Regex.Matches(input, pattern2))
+         Console.WriteLine("'{0}' found at position {1}", 
+                           match.Value, match.Index);
+   }
+}
+// The example displays the following output:
+//       'gray' found at position 4
+//       'grey' found at position 35
+//       
+//       'gray' found at position 4
+//       'grey' found at position 35           
+```
+
+```vb
+Imports System.Text.RegularExpressions
+
+Module Example
+   Public Sub Main()
+      ' Regular expression using character class.
+      Dim pattern1 As String = "\bgr[ae]y\b"
+      ' Regular expression using either/or.
+      Dim pattern2 As String = "\bgr(a|e)y\b"
+
+      Dim input As String = "The gray wolf blended in among the grey rocks."
+      For Each match As Match In Regex.Matches(input, pattern1)
+         Console.WriteLine("'{0}' found at position {1}", _
+                           match.Value, match.Index)
+      Next      
+      Console.WriteLine()
+      For Each match As Match In Regex.Matches(input, pattern2)
+         Console.WriteLine("'{0}' found at position {1}", _
+                           match.Value, match.Index)
+      Next      
+   End Sub
+End Module
+' The example displays the following output:
+'       'gray' found at position 4
+'       'grey' found at position 35
+'       
+'       'gray' found at position 4
+'       'grey' found at position 35 
+```
+
+The regular expression that uses the | character, `\bgr(a|e)y\b,` is interpreted as shown in the following table.
+
+Pattern | Description
+------- | -----------
+`\b` | Start at a word boundary.
+`gr` | Match the characters "gr".
+`(a&#124;e)` | Match either an "a" or an "e".
+`y\b` |	Match a "y" on a word boundary.
+
+
+The | character can also be used to perform an either/or match with multiple characters or subexpressions, which can include any combination of character literals and regular expression language elements. (The character class does not provide this functionality.) The following example uses the | character to extract either a U.S. Social Security Number (SSN), which is a 9-digit number with the format *ddd-dd-dddd*, or a U.S. Employer Identification Number (EIN), which is a 9-digit number with the format *dd-ddddddd*.
+
+```csharp
```

Comment:
My preference would be to fix this here and open an issue to fix this in the others. It makes it really hard to read to have both code languages in the topic